odoo/addons/base/models/ir_attachment.py

```diff
@@ -814,3 +792,20 @@ def _to_http_stream(self):
             stream.size = 0
 
         return stream
+
+    def _can_elevate_access(self, access_token, field):
+        record_sudo = self.sudo()
+        if access_token:
+            tok = record_sudo.with_context(prefetch_fields=False).access_token
+            valid_token = consteq(tok or "", access_token)
+            if not valid_token:
+                raise AccessError("Invalid access token")
+            return True
+        if record_sudo.with_context(prefetch_fields=False).public:
```

Comment:
looks like `.with_context(prefetch_fields=False)` can be moved to `record_sudo` first line then
but it seems pretty counter-productive to fetch each of `access_token` and `public` separately.
It might be worth just calling an explicit fetch on them at the start instead.